### ISSUE TITLE
Unable to build Bugsnag 5.23.0 for tvOS using Carthage

### ISSUE THREAD

**Issue by faithfracture** [bugsnag/bugsnag-cocoa#523]
### Description
I'm trying to update Bugsnag-cocoa from 5.22.6 to 5.23.0 and it fails to build. 5.22.6, .7, .8 all fail for all platforms completely. 5.22.9 and 5.23.0 build for iOS, but that's it. Judging from the log output it looks like the build scheme isn't configured correctly - all of the tvOS simulators are listed as "Ineligible destinations"

### Issue
It fails to build

### Environment
Xcode 11.4

Library versions:

* Bugsnag version (from your Podfile, Podfile.lock, or elsewhere): 5.23.0
* Carthage version (if any): 0.34.0

### Example code snippet
Cartfile:

```
github "AFNetworking/AFNetworking" ~> 4.0
github "jdg/MBProgressHUD" ~> 1.1
github "erikdoe/ocmock" ~> 3.4
github "bugsnag/bugsnag-cocoa"
```

<details><summary>Error messages:</summary>

Command Line
```
 $ ✗  carthage build --platform tvos
*** xcodebuild output can be found in /var/folders/vx/yjcdkqmd4x9_6yc851v4ch9m0000gn/T/carthage-xcodebuild.p187R4.log
*** Downloading ocmock.framework binary at "OCMock 3.6"
*** Building scheme "AFNetworking tvOS" in AFNetworking.xcworkspace
*** Building scheme "Bugsnag" in Bugsnag.xcodeproj
*** Building scheme "Bugsnag" in Bugsnag.xcodeproj
Build Failed
	Task failed with exit code 70:
	/usr/bin/xcrun xcodebuild -project /project/Carthage/Checkouts/bugsnag-cocoa/iOS/Bugsnag.xcodeproj -scheme Bugsnag -configuration Release -derivedDataPath /Users/myuser/Library/Caches/org.carthage.CarthageKit/DerivedData/11.4_11E146/bugsnag-cocoa/v5.23.0 -sdk appletvsimulator -destination platform=tvOS\ Simulator,id=5E249F73-CE56-4285-956B-6DFBE09DD3E2 -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /project/Carthage/Checkouts/bugsnag-cocoa)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/vx/yjcdkqmd4x9_6yc851v4ch9m0000gn/T/carthage-xcodebuild.p187R4.log
```

Message in `/var/folders/vx/yjcdkqmd4x9_6yc851v4ch9m0000gn/T/carthage-xcodebuild.p187R4.log`
```
User defaults from command line:
    IDEDerivedDataPathOverride = /Users/jordanbondo/Library/Caches/org.carthage.CarthageKit/DerivedData/11.4_11E146/bugsnag-cocoa/v5.23.0

Build settings from command line:
    CARTHAGE = YES
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = appletvsimulator13.4

xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ platform:tvOS Simulator, id:5E249F73-CE56-4285-956B-6DFBE09DD3E2 }

	Available destinations for the "Bugsnag" scheme:
		{ platform:macOS, arch:x86_64, id:C7EA8FC6-2E8E-59BE-989D-23D58D5C43BD }
		{ platform:macOS, arch:x86_64, variant:Mac Catalyst, id:C7EA8FC6-2E8E-59BE-989D-23D58D5C43BD }
		{ platform:iOS, id:00008020-001E14AA2241002E, name:Joruus }
		{ platform:iOS, id:136f933879c8b08375c62acc89062c99b10c0538, name:QA iPhone X }
		{ platform:iOS Simulator, id:71EB2F47-3C68-4FD4-9AB3-5A4C8E694DEB, OS:13.4, name:iPad (7th generation) }
		{ platform:iOS Simulator, id:07894517-B488-426F-8FDA-E440209CEA96, OS:13.4, name:iPad Air (3rd generation) }
		{ platform:iOS Simulator, id:E1604C0D-6192-452C-8179-18B7A637B45D, OS:13.4, name:iPad Pro (9.7-inch) }
		{ platform:iOS Simulator, id:0EDB4919-ECF8-4484-9417-9E2956F6C3A9, OS:13.4, name:iPad Pro (11-inch) (2nd generation) }
		{ platform:iOS Simulator, id:F0005707-90A6-4057-A3F4-C64622426A5A, OS:13.4, name:iPad Pro (12.9-inch) (4th generation) }
		{ platform:iOS Simulator, id:805CEA89-A891-4258-969A-CD89A51BB320, OS:13.4, name:iPhone 8 }
		{ platform:iOS Simulator, id:EDBFAFCC-46DF-4210-B0F8-378CF6F5B6FC, OS:13.4, name:iPhone 8 Plus }
		{ platform:iOS Simulator, id:88411655-F5DD-4410-9660-E512F6800D5E, OS:13.4, name:iPhone 11 }
		{ platform:iOS Simulator, id:3CC41C78-0F68-4B91-9420-2160896E8A57, OS:13.4, name:iPhone 11 Pro }
		{ platform:iOS Simulator, id:372A6BBE-8373-4292-A7F8-FAA7F7286301, OS:13.4, name:iPhone 11 Pro Max }

	Ineligible destinations for the "Bugsnag" scheme:
		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Generic iOS Device }
		{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Generic iOS Simulator Device }
		{ platform:tvOS, id:dvtdevice-DVTiOSDevicePlaceholder-appletvos:placeholder, name:Generic tvOS Device }
		{ platform:tvOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-appletvsimulator:placeholder, name:Generic tvOS Simulator Device }
		{ platform:watchOS, id:dvtdevice-DVTiOSDevicePlaceholder-watchos:placeholder, name:Generic watchOS Device }
		{ platform:watchOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-watchsimulator:placeholder, name:Generic watchOS Simulator Device }
```
</details>

**Comment by xander-jones** [comment]
Hey @faithfracture, can you please try running `carthage update --platform tvos` so that you generate the frameworks to add to your project from the repositories specified in the Cartfile?

**Comment by faithfracture** [comment]
No change, same result.

**Comment by xander-jones** [comment]
Hey @faithfracture we've managed to reproduce this, and are taking a look into a fix. We'll keep you posted on here.

**Comment by faithfracture** [comment]
Awesome! Thanks 👍

**Comment by robinmacharg** [comment]
Hi @faithfracture, just to keep you updated: You'll see the proposed fix for Carthage support above, in #535.  This hasn't been merged yet due to some lingering internal discussion over Mac Catalyst support.  However, if you'd like to point your Cartfile directly at the PR commit, you should now be able to `carthage update` (with or without `--platform tvOS`) and find a build for tvOS in the appropriate Build directory.  Specifically, add the following line to your Cartfile: 

```
github "bugsnag/bugsnag-cocoa" "764ddb4"
```
Drag the framework into your XCode project (remembering to set it to "Embed & Sign" or "Embed Without Signing" in the project/target's General tab) and add the appropriate `import` and `[Bugsnag init:]` code as per the docs.

We'll update this issue once the fix is merged.

**Comment by faithfracture** [comment]
Thanks! I'm going to be working on my tvOS project here in about a week, so I'll give that a go when I'm updating the rest of my dependencies.

**Comment by mattdyoung** [comment]
Fixed in [v6.0.0](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.0.0).

Note: There are a number of breaking changes in v6.0.0 as part of an effort to improve the interface and make it more consistent across our error reporting libraries, so please refer to the upgrade guide:
https://github.com/bugsnag/bugsnag-cocoa/blob/master/UPGRADING.md
